src/client/activeobjectmgr.cpp

```diff
@@ -28,11 +28,15 @@ namespace client
 void ActiveObjectMgr::clear()
 {
 	// delete active objects
+
+	// This loop is required, as ~ClientActiveObject() can access m_active_objects,
+	// which is invalid while m_active_objects.clear() is being executed.
 	for (auto &active_object : m_active_objects) {
-		delete active_object.second;
-		// Object must be marked as gone when children try to detach
-		active_object.second = nullptr;
+		// By resetting to null, the object is also marked as gone, which is
+		// required when children try to detach.
+		active_object.second.reset();
```

Comment:
the (implicit) iterator could also be invalidated while this runs, in fact
honestly there are way too many pitfalls here.